### PR TITLE
Added function remove_value to HashMapContext

### PR DESCRIPTION
**Pull request by SudoDios**
Implements a new method to remove variables from context maps.
Returns the removed Value if present, or None otherwise.
Includes corresponding unit test for verification.

-------------

* [x] I publish this contribution under the [MIT License](https://opensource.org/license/mit).

### CODE REVIEW

**Comment by SudoDios** [comment]
Oh, yes, definitely.

**Comment by SudoDios** [comment]
Coverage requires that the default body be tested as well.
Is it necessary?

**Comment by sebschmi** [comment]
Nah it's fine. Thanks for the contribution!

File: tests/integration.rs [hunk 1]
```diff
@@ -142,6 +142,9 @@ fn test_with_context() {
         eval_with_context("five < six && true", &context),
         Ok(Value::Boolean(true))
     );
+    assert_eq!(context.remove_value("half"), Ok(Some(Value::Float(0.5))));
+    assert_eq!(context.remove_value("zero"), Ok(Some(Value::Int(0))));
+    assert_eq!(context.remove_value("zero"), Ok(None));
```
Comment (sebschmi) [inline tests/integration.rs]: Let's also test here that the value is gone when evaluating, something like: `assert_eq!(eval_with_context("zero", &context), Err(...));`

**Review by sebschmi — CHANGES_REQUESTED** [review]
Just some minor things, would be great if you could fix them.
